fe/fe-core/src/main/java/org/apache/doris/catalog/ScalarType.java

```diff
@@ -69,6 +66,12 @@
     // Keep consistent with backend ColumnType::CHAR_INLINE_LENGTH
     public static final int CHAR_INLINE_LENGTH = 128;
 
+    // Max length of String types, in be storage layer store string length
+    // using int32, the max length is 2GB, the first 4 bytes store the length
+    // so the max available length is 2GB - 4
+
+    public static final int MAX_STRING_LENGTH = 0x7fffffff - 4;
```

Comment:
I think `public static final int MAX_STRING_LENGTH = 2 << 30 - 1 - 4;` is more clear here. 
And you should explain why did you minus 1 here.

Reply:
It is explained in comments that the value is 2GB - 4, why  use `public static final int MAX_STRING_LENGTH = 2 << 30 - 1 - 4;` ?  How to compute 2GB - 4 is not important and  2 << 30 does not have performance advantages